shell.nix

```diff
@@ -150,6 +150,8 @@ haskell.project.shellFor {
   withHoogle = false;
 
   shellHook = ''
+    export LC_ALL=C.utf8
```

Comment:
This looks harmless but might have subtle effects, as it kind of affects *everything* inside the shell.
Inside the nix-shell we only have `C`, `C.utf8` and `POSIX` locales.
Some users (for example those with `LC_CTYPE="en_AU.UTF-8"`) will not be able to run `cabal test plutus-contract` as it will break when printing non-ascii chars to stdout.
Adding this line to the shell fixes the problem, but I'm a little concerned it might blow up in our faces at some point.
What do you think? @michaelpj @angerman @koslambrou

Reply:
Are you maybe missing this line? https://github.com/input-output-hk/plutus/blob/master/shell.nix#L113

This is kind of my fault - `haskell.nix` used to set this stuff in the shell automatically, but I removed it because no other nixpkgs shells set this stuff, we usually leave it to the user. But it seems to be needed for a bunch of Haskell stuff.

Reply:
To be clear, I think including the locale archives should remove the need for this setting. Overriding the user's locale can be annoying for them.

Reply:
Aaaa so `plutus` has it but `plutus-apps` does not. Let's try that. @andreabedini could you test out this branch and tell me if it works?

Reply:
@zeme-iohk yes I will test. Adding the missing locale information sounds like a clearner fix, I wasn't aware that was a solved problem already.